src/main/java/com/fauna/feed/FeedPage.java

```diff
@@ -0,0 +1,151 @@
+package com.fauna.feed;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fauna.codec.Codec;
+import com.fauna.exception.ClientResponseException;
+import com.fauna.response.QueryStats;
+import com.fauna.response.StreamEvent;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.http.HttpResponse;
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.fasterxml.jackson.core.JsonToken.END_ARRAY;
+import static com.fasterxml.jackson.core.JsonToken.FIELD_NAME;
+import static com.fasterxml.jackson.core.JsonToken.START_ARRAY;
+import static com.fasterxml.jackson.core.JsonToken.START_OBJECT;
+import static com.fauna.constants.ResponseFields.EVENTS_FIELD_NAME;
+import static com.fauna.constants.ResponseFields.FEED_HAS_NEXT_FIELD_NAME;
+import static com.fauna.constants.ResponseFields.STATS_FIELD_NAME;
+import static com.fauna.constants.ResponseFields.CURSOR_FIELD_NAME;
+
+public class FeedPage<E> {
```

Comment:
I think this class and FeedRequest might deserve some unit tests because of the token logic, but I'm fine if that comes later.

Reply:
Agree.